File: ios/MagicFormsMobile/Config/APIConfig.swift
import Foundation

/// API base URL for the MagicForms backend (main domain only — no entity slug in paths).
///
/// Set once at build time via the `API_BASE_URL` key in Info.plist (see Xcode target build settings).
/// End users never choose or enter a server URL.
enum APIConfig {
    static var baseURL: URL {
        if let raw = Bundle.main.object(forInfoDictionaryKey: "API_BASE_URL") as? String,
           !raw.isEmpty,
           let url = URL(string: raw) {
            return url
        }
        #if DEBUG
        #if targetEnvironment(simulator)
        return URL(string: "http://127.0.0.1:8000")!
        #else
        // Physical device in Debug: set API_BASE_URL in the target’s build settings (e.g. your Mac’s LAN IP).
        return URL(string: "http://127.0.0.1:8000")!
        #endif
        #else
        fatalError("Set API_BASE_URL in the MagicFormsMobile target build settings for Release builds.")
        #endif
    }

    static func url(path: String) -> URL {
        var base = baseURL.absoluteString
        if base.hasSuffix("/") { base.removeLast() }
        let suffix = path.hasPrefix("/") ? path : "/\(path)"
        guard let url = URL(string: base + suffix) else {
            fatalError("Invalid API URL: \(base)\(suffix)")
        }
        return url
    }
}

File: android/app/src/main/java/com/magicforms/mobile/data/api/ApiConfig.kt
package com.magicforms.mobile.data.api

import com.magicforms.mobile.BuildConfig

/** Main-site API base URL (no entity slug in paths). Set via [BuildConfig.API_BASE_URL]. */
object ApiConfig {
    val baseUrl: String
        get() = BuildConfig.API_BASE_URL.trimEnd('/')

    fun url(path: String): String {
        val suffix = if (path.startsWith("/")) path else "/$path"
        return baseUrl + suffix
    }

    fun documentUrl(apiPath: String, inline: Boolean = false): String {
        val base = url(apiPath)
        return if (inline) "$base?inline=1" else base
    }
}
